client/src/app/auth/reset-password/page.tsx

```diff
@@ -0,0 +1,167 @@
+"use client";
+import { useState, useEffect } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import useFetch from "@/src/hooks/useFetch";
+import { AuthService } from "@/src/services/AuthService";
+import Navbar from "@/src/components/Navbar/Navbar";
+import Footer from "@/src/components/Footer/Footer";
+import useThemeStore from "@/src/store/themeStore";
+import Image from "next/image";
+import ResetPasswordSvg from "@/public/assets/Login.svg";
+import Link from "next/link";
+
+const ResetPassword = () => {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const theme = useThemeStore((state) => state.theme);
+  const [token, setToken] = useState<string | null>(null);
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+  const [successMessage, setSuccessMessage] = useState("");
+
+  const [{ isLoading, data, isError, error }, resetPasswordAPI] = useFetch(null);
+
+  useEffect(() => {
+    const token = searchParams.get("token");
+    if (token) {
+      setToken(token);
+    }
```

Comment:
**Bug**
If token is not present, then this page will not open. Just redirect to other login page if token is not present.